### ISSUE TITLE
Compress-Archive creates archives some systems can't process (missing directory objects?)

### ISSUE THREAD

**Issue by sba923** [PowerShell/Microsoft.PowerShell.Archive#85]
# Steps to reproduce

```
Compress-Archive -LiteralPath d:\tmp\somedir -DestinationPath d:\tmp\somearchive.zip
```

# Expected behavior

The resulting archive should have the following contents (as displayed by '7z.exe l')

```
7-Zip 19.00 (x64) : Copyright (c) 1999-2018 Igor Pavlov : 2019-02-21

Scanning the drive for archives:
1 file, 475290 bytes (465 KiB)

Listing archive: D:\tmp\somedir-7z.zip

--
Path = D:\tmp\somedir-7z.zip
Type = zip
Physical Size = 475290

   Date      Time    Attr         Size   Compressed  Name
------------------- ----- ------------ ------------  ------------------------
2019-09-14 19:45:09 D....            0            0  somedir
2019-09-11 21:52:37 ....A       489436       474779  somedir\fintech - Wikipedia - EN.pdf
2018-09-01 08:08:28 ....A           15           15  somedir\someotherfile.txt
------------------- ----- ------------ ------------  ------------------------
2019-09-14 19:45:09             489451       474794  2 files, 1 folders

```
(this is an archive created by 7-Zip, not by Windows Explorer)

# Actual behavior

The archive created by `Compress-Archive` has the following contents:

```
7-Zip 19.00 (x64) : Copyright (c) 1999-2018 Igor Pavlov : 2019-02-21

Scanning the drive for archives:
1 file, 475247 bytes (465 KiB)

Listing archive: D:\tmp\somedir-PS.zip

--
Path = D:\tmp\somedir-PS.zip
Type = zip
Physical Size = 475247

   Date      Time    Attr         Size   Compressed  Name
------------------- ----- ------------ ------------  ------------------------
2019-09-11 21:52:36 .....       489436       474934  somedir\fintech - Wikipedia - EN.pdf
2018-09-01 08:08:28 .....           15           17  somedir\someotherfile.txt
------------------- ----- ------------ ------------  ------------------------
2019-09-11 21:52:36             489451       474951  2 files
```
Note the 'blank' attributes, and the lack of a directory entry.

This causes such an archive to be rejected by some applications (Kodi to be specific).

Note that this is not PS7-specific, I get the same results with PS6.2.3 or Windows PowerShell 5.1

With a larger test set with many subdirectories, the archive created by `Compress-Archive` contains only one directory object, all other directories in the source tree are missing from the archive.

# Environment data

```
Name                           Value
----                           -----
PSVersion                      7.0.0-preview.4
PSEdition                      Core
GitCommitId                    7.0.0-preview.4
OS                             Microsoft Windows 10.0.18362
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

````
PS 7.0.0-preview.4 //pnjnas/pi/kodi> gcm compress-archive

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Function        Compress-Archive                                   1.2.3.0    Microsoft.PowerShell.Archive
````

**Comment by ayousuf23** [comment]
This should be fixed.

**Comment by sba923** [comment]
I'm trying to test, but I can't find how to install the 2.x preview next to 1.x:

````powershell
PS> gcm Compress-Archive

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Function        Compress-Archive                                   1.2.5      Microsoft.PowerShell.Archive

PS> Install-Module -Name Microsoft.PowerShell.Archive -AllowPrerelease -force -ver -scope
 currentuser
VERBOSE: Using the provider 'PowerShellGet' for searching packages.
VERBOSE: The -Repository parameter was not specified.  PowerShellGet will use all of the registered repositories.
VERBOSE: Getting the provider object for the PackageManagement Provider 'NuGet'.
VERBOSE: The specified Location is 'https://www.powershellgallery.com/api/v2' and PackageManagementProvider is 'NuGet'.
VERBOSE: Searching repository 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='Microsoft.PowerShell.Archive'' for ''.
VERBOSE: Total package yield:'1' for the specified package 'Microsoft.PowerShell.Archive'.
VERBOSE: Performing the operation "Install-Module" on target "Version '2.0.1-preview2' of module 'Microsoft.PowerShell.Archive'".
VERBOSE: The installation scope is specified to be 'currentuser'.
VERBOSE: The specified module will be installed in 'C:\Users\<REDACTED>\OneDrive\Documents\PowerShell\Modules'.
VERBOSE: The specified Location is 'NuGet' and PackageManagementProvider is 'NuGet'.
VERBOSE: Downloading module 'Microsoft.PowerShell.Archive' with version '2.0.1-preview2' from the repository 'https://www.powershellgallery.com/api/v2'.
VERBOSE: Searching repository 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='Microsoft.PowerShell.Archive'' for ''.
VERBOSE: InstallPackage' - name='Microsoft.PowerShell.Archive', version='2.0.1-preview2',destination='C:\Users\<REDACTED>\AppData\Local\Temp\1431991098'
VERBOSE: DownloadPackage' - name='Microsoft.PowerShell.Archive', version='2.0.1-preview2',destination='C:\Users\<REDACTED>\AppData\Local\Temp\1431991098\Microsoft.PowerShell.Archive.2.0.1-preview2\Microsoft.PowerShell.Archive.2.0.1-preview2.nupkg', uri='https://www.powershellgallery.com/api/v2/package/Microsoft.PowerShell.Archive/2.0.1-preview2'
VERBOSE: Downloading 'https://www.powershellgallery.com/api/v2/package/Microsoft.PowerShell.Archive/2.0.1-preview2'.
VERBOSE: Completed downloading 'https://www.powershellgallery.com/api/v2/package/Microsoft.PowerShell.Archive/2.0.1-preview2'.
VERBOSE: Completed downloading 'Microsoft.PowerShell.Archive'.
VERBOSE: InstallPackageLocal' - name='Microsoft.PowerShell.Archive', version='2.0.1-preview2',destination='C:\Users\<REDACTED>\AppData\Local\Temp\1431991098'
VERBOSE: Validating the 'Microsoft.PowerShell.Archive' module contents under 'C:\Users\<REDACTED>\AppData\Local\Temp\1431991098\Microsoft.PowerShell.Archive.2.0.1-preview2' path.
VERBOSE: Test-ModuleManifest successfully validated the module manifest file 'C:\Users\<REDACTED>\AppData\Local\Temp\1431991098\Microsoft.PowerShell.Archive.2.0.1-preview2'.
VERBOSE: Validating the authenticode signature and publisher of the catalog file or module manifest file of the module 'Microsoft.PowerShell.Archive'.
VERBOSE: Catalog file 'Microsoft.PowerShell.Archive.cat' is not found in the contents of the module 'Microsoft.PowerShell.Archive' being installed.
VERBOSE: Valid authenticode signature found in the file 'Microsoft.PowerShell.Archive.psd1' for the module 'Microsoft.PowerShell.Archive'.
VERBOSE: For publisher validation, current module 'Microsoft.PowerShell.Archive' with version '2.0.1' with publisher name 'CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US' from root certificate authority 'CN=Microsoft Root Certificate Authority 2011, O=Microsoft Corporation, L=Redmond, S=Washington, C=US'. Is this module signed by Microsoft: 'True'.
VERBOSE: For publisher validation, using the previously-installed module 'Microsoft.PowerShell.Archive' with version '1.2.5' under 'C:\program files\powershell\7\Modules\Microsoft.PowerShell.Archive' with publisher name 'CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US' from root certificate authority 'CN=Microsoft Root Certificate Authority 2011, O=Microsoft Corporation, L=Redmond, S=Washington, C=US'. Is this module signed by Microsoft: 'True'.
VERBOSE: Authenticode issuer 'CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US' of the new module 'Microsoft.PowerShell.Archive' with version '2.0.1' matches with the authenticode issuer 'CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US' of the previously-installed module 'Microsoft.PowerShell.Archive' with version '1.2.5'.
VERBOSE: Checking for possible command collisions for the module 'Microsoft.PowerShell.Archive' commands.
Install-Package: Administrator rights are required to install or update. Log on to the computer with an account that has Administrator
rights, and then try again, or install by adding "-Scope CurrentUser" to your command. You can also try running the
Windows PowerShell session with elevated rights (Run as Administrator).
````

(same error from an elevated PowerShell prompt!)

What am I missing?

**Comment by sba923** [comment]
Now that @SteveL-MSFT has indicated via https://twitter.com/Steve_MSFT/status/1560713898159771648?s=20&t=lvJT9B0xwzt0RLgtRM3AgA that Microsoft.PowerShell.Archive _currently_ requires PowerShell 7.3, I've tested with 7.3.0-preview.7 that the problem's fixed in 2.0.1-preview2:

<img width="784" alt="image" src="https://user-images.githubusercontent.com/12860484/185699635-342029c5-1e31-4f74-b9c2-8f82748f892f.png">
